unattended-upgrade

```diff
@@ -1487,12 +1519,70 @@ def main(options, rootdir=""):
     if rootdir:
         _setup_alternative_rootdir(rootdir)
 
+    # see debian #776752
+    install_start_time = datetime.datetime.now().replace(microsecond=0)
+
     # setup logging
     mem_log = _setup_logging(options)
+    # get log
+    logfile_dpkg = os.path.join(_get_logdir(), 'unattended-upgrades-dpkg.log')
+    if not os.path.exists(logfile_dpkg):
+        with open(logfile_dpkg, 'w'):
+            pass
+
+    # lock for the shutdown check
+    shutdown_lock = apt_pkg.get_lock(LOCK_FILE)
+    if shutdown_lock < 0:
+        logging.error("Lock file is already taken, exiting")
+        return 1
+
+    try:
+        res = main_inner(options, rootdir, mem_log, logfile_dpkg,
+                         install_start_time)
+        if res.result_str and not options.dry_run:
+            # there is some meaningful result which is worth an email
+            log_content = get_dpkg_log_content(logfile_dpkg,
+                                               install_start_time)
+            send_summary_mail(res.pkgs, res.success, res.result_str,
+                              res.pkgs_kept_back, res.pkgs_removed,
+                              res.pkgs_kept_installed, mem_log,
+                              log_content)
+        if (res.update_stamp):
+            # write timestamp file
+            write_stamp_file()
+            if not options.dry_run:
+                # check if the user wants a reboot
+                reboot_if_requested_and_needed()
+        os.close(shutdown_lock)
+        if res.success:
+            return 0
+        else:
+            return 1
+
+    except Exception as e:
```

Comment:
You could probably use  `sys.excepthook` instead of splitting main/main_innter but its fine this way.